### PR TITLE
Add test: "Unable to create a flow on an isl port"

### PR DESCRIPTION
**Pull request by andriidovhan**
Add test: "Unable to create a flow on an isl port"

_How to find a flow with a conflict port?_
```
MATCH ()-[i:isl]->()
MATCH ()-[f:flow]->() 
where f.src_switch = i.src_switch and f.src_port = i.src_port
return f.src_switch as SwitchId, f.flowid as flowId, f.src_port as PortNumber
```
Close:  #2222

### CODE REVIEW

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeSpec.groovy [hunk 1]
```diff
@@ -117,6 +118,24 @@ class FlowValidationNegativeSpec extends BaseSpecification {
         "synchronize" | "Could not reroute flow: Flow $NON_EXISTENT_FLOW_ID not found"
     }
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeSpec.groovy]: Should be in FlowCrudSpec

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeSpec.groovy [hunk 2]
```diff
@@ -117,6 +118,24 @@ class FlowValidationNegativeSpec extends BaseSpecification {
...
 
+    def "Unable to create a flow on an isl port"() {
+        given: "A isl"
+        Isl isl = topology.islsForActiveSwitches.find { it.aswitch && it.dstSwitch }
+        assert isl
+
+        when: "Try to create a flow using isl port"
+        def flow = flowHelper.randomFlow(isl.srcSwitch, isl.dstSwitch)
+        flow.source.portNumber = isl.srcPort
+        flow.destination.portNumber = isl.dstPort
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeSpec.groovy]: Please make 2 iterations. 1 for src violation and 1 for dst violation

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeSpec.groovy]: done

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeSpec.groovy (hunk 2 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeSpec.groovy]: done.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy [hunk 3]
```diff
@@ -437,6 +438,82 @@ class FlowCrudSpec extends BaseSpecification {
         }
     }
 
+    @Unroll
+    def "Unable to create a flow on an isl port in case port is occupied by a #data.switchType switch"() {
```
[the remark this answers is not in the document]
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: fixed

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy [hunk 4]
```diff
@@ -437,6 +438,82 @@ class FlowCrudSpec extends BaseSpecification {
...
+                        port      : "srcPort",
+                        messgae   : { Isl violatedIsl ->
+                            "Could not update flow: The port $violatedIsl.srcPort " +
+                                    "on the switch '$violatedIsl.srcSwitch.dpId' is occupied by an ISL."
+                        }
+                ],
+                [
+                        switchType: "destination",
+                        port      : "dstPort",
+                        messgae   : { Isl violatedIsl ->
```
[the remark this answers is not in the document]
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: fixed

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy [hunk 5]
```diff
@@ -437,6 +438,82 @@ class FlowCrudSpec extends BaseSpecification {
...
+                            "Could not update flow: The port $violatedIsl.dstPort " +
+                                    "on the switch '$violatedIsl.dstSwitch.dpId' is occupied by an ISL."
+                        }
```
[the remark this answers is not in the document]
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy]: done.
the spec level closure  was added

**Pull request merged**